### DISCUSSION TITLE
New sampler that extends CutSampler

### DISCUSSION THREAD

**Discussion by desh2608** [lhotse-speech/lhotse#861] [Q&A] [answered]
For a project that I'm working on that uses Lhotse for the data pipeline, I want to extend Lhotse's sampler to apply some additional constraints on the batches (mainly to enforce that each batch should have cuts from the same speaker). What would be the most straight-forward way to implement this?

**Comment by pzelasko** [comment, marked as answer]
I suggest to use RoundRobinSampler that consists of a list of DynamicBucketingSampler instances, one for each speaker. In case the speakers have a very small amount of data, use DynamicCutSampler instead. Would that work?

**Reply by desh2608** [reply]
Thanks! That's exactly what I was looking for.
